modules/core/src/lib/layer.js

```diff
@@ -662,6 +664,13 @@ ${flags.viewportChanged ? 'viewport' : ''}\
     return this.setChangeFlags(changeFlags);
   }
 
+  setModuleParameters(moduleParameters) {
+    if (this.internalState) {
+      Object.assign(this.internalState.moduleParameters, moduleParameters);
```

Comment:
Seeing it like this does seem to duplicate the intention of the system. If the app has to do this anyway, shouldn't we just move this into the module system?